Map/RoomUI.gd

```diff
@@ -1,7 +1,39 @@
 extends Control
+class_name RoomUI
 
 var room: RoomBase
 @export var label: Label
+@export var color_rect: ColorRect
+@export var player_icon: TextureRect
+
+var floor_index: int = 0
+var room_index: int = 0
+
+func _ready():
+	if room != null:
+		if !room.on_light_level_changed.is_connected(_update_room_UI):
+			room.on_light_level_changed.connect(_update_room_UI)
+		_update_room_UI()
 
 func set_label(title: String):
 	label.set_text(title)
+
+func set_player_icon(enabled: bool) -> void:
```

Comment:
activate instead of set maybe ? this name makes it feel like i'm gonna change the player icon when what it does is make it appear on screen
activate maybe not the best, because it could be with enabled = false
`toggle_player_icon` ?